### DISCUSSION TITLE
open_mfdataset error: var not present in all datasets or conflicting dimensions

### DISCUSSION THREAD

**Discussion by gewitterblitz** [pydata/xarray#7666] [Q&A] [answered]
Hi,

I have unsuccessfully tried loading multiple netcdf files using `xr.open_mfdataset`. The issue is that not all the netcdf files contain the desired variable in them. Secondly, the dimension `cells` does not have the same size across all netcdf files. Here's a few screenshots of individual datasets for reference (notice the variable dimension size along dim ='cells'):

<img width="746" alt="Screenshot 2023-03-23 at 1 13 42 PM" src="https://user-images.githubusercontent.com/13985417/227307822-dc590087-f999-4ef5-a591-6c2713b0fc2f.png">

<img width="715" alt="Screenshot 2023-03-23 at 1 13 06 PM" src="https://user-images.githubusercontent.com/13985417/227307532-7c5d824a-dcf7-4f58-a8eb-eae2fade28cd.png">

<img width="729" alt="Screenshot 2023-03-23 at 1 13 20 PM" src="https://user-images.githubusercontent.com/13985417/227307670-ef4f6f10-70b1-472d-9d1a-ce437df51833.png">

Here's what I have tried so far:

```
def preprocessing(ds):
    if ds['cells'].size == 0:
        ds_new = ds.assign_coords({'cell':np.nan})
        return ds_new
    else:
        return ds
    
xr.open_mfdataset(files,combine='nested',concat_dim='cells',preprocess=preprocessing,data_vars='different')
```
`ValueError: 'corrected_differential_reflectivity' is not present in all datasets.`

```
xr.open_mfdataset(files,combine='nested',concat_dim='iops',preprocess=preprocessing)
```

`ValueError: cannot reindex or align along dimension 'cells' because of conflicting dimension sizes: {1, 2, 3, 4}`

**Comment by dcherian** [comment, marked as answer]
I think we should've fixed this on xarray 2023.02.0 https://github.com/pydata/xarray/pull/7400 . What version are you running?

cc @kmuehlbauer

**Comment by gewitterblitz** [comment]
Thanks, @dcherian, the new version works fine. My conda environment took forever to update xarray (and other dependencies) from version '2023.1.0'.
